extension/services/deezer/player.js

```diff
@@ -1,52 +1,68 @@
 /* globals helpers */
 
 this.player = (function() {
-  const SEARCH_PLAY = ".datagrid-row song a[role='button']";
+  const SEARCH_PLAY = ".datagrid-row a[role='button']";
 
   class Player extends helpers.Runner {
     action_play() {
       const button = this.querySelector("button[aria-label='Play']");
       button.click();
     }
 
-    async action_search({ query, thenPlay }) {
+    async search(query) {
       try {
+        const queryInput = await this.waitForSelector(
+          "form .topbar-search-input"
+        );
         const querySubmit = this.querySelector("form .topbar-search-submit");
-        const queryInput = this.querySelector("form .topbar-search-input");
         this.setReactInputValue(queryInput, query);
-        queryInput.value = query;
         querySubmit.click();
-        if (thenPlay) {
-          const playerButton = await this.waitForSelector(SEARCH_PLAY, {
-            timeout: 2000,
-            // There seem to be 3 fixed buttons that appear early before the search results
-            minCount: 4,
-          });
-          playerButton.click();
+      } catch {
+        throw new Error("Please log in to use this service.");
+      }
+    }
+
+    async action_search({ query, thenPlay }) {
+      try {
+        try {
+          await this.search(query);
+          if (thenPlay) {
+            const playerButton = await this.waitForSelector(SEARCH_PLAY, {
+              timeout: 2000,
+            });
+            playerButton.click();
+          }
+        } catch (e) {
+          throw new Error("No Search Results!");
         }
       } catch (e) {
-        const unlogged = this.querySelector("div[class='unlogged-homepage']");
-        if (unlogged) {
-          throw new Error("Please log in to use this service.");
-        }
+        throw new Error(e.message);
```

Comment:
This doesn't do anything any longer (except break the traceback by recreating the exception). But it also looks like this code isn't right. It should probably be:

```js
...} catch (e) {
  const unlogged = document.querySelector(".unlogged-homepage");
  if (unlogged) {
    throw new Error("Please log in to use this service.");
  }
  throw e;
}
```

Reply:
If it always re-throws the error, then the try/catch isn't necessary at all.